### DISCUSSION TITLE
How to setup code compilation in STM32CubeIDE for MHP30?

### DISCUSSION THREAD

**Discussion by corndog2000** [Ralim/IronOS#1994] [Q&A] [answered]
I'm trying to set up the STM32CubeIDE for use in compiling IronOS for an MHP30 hotplate. I cannot figure out the required project includes. Does someone already have the project file setup that I could download? If not, could someone help me out in what my project includes should look like? I'm looking for specific included paths to add to the project configuration.

**Comment by ia** [comment, marked as answer]
Hello. [Here is the only one brief mention](https://ralim.github.io/IronOS/Development/#stm32cubeide) of setting up _STM32CubeIDE_, from the official documentation.

The IronOS source tree doesn't rely on any **external** third party libraries (_FreeRTOS_ code is embedded into the tree and `usb-pd` lies inside of the tree as _git submodule_). So everything you need is inside the tree. Plus, you can use `source/Makefile` as a reference to get the idea of dependencies between project directories, source files & include files.

I can speak only for myself here, but as far as I know, we all just use our favorite text editors/IDEs/tools configured for the source tree, and we just fire up `make ...` when we need to ~~build~~ _deal with the consequences_. :)

It benefits since there is no ide/tooling vendor lock-in, which gives huge abilities for autobuilding & autotesting.
